lib/erb_lint/corrector.rb

```diff
@@ -17,11 +17,22 @@ def corrections
     end
 
     def corrector
-      RuboCop::Cop::Corrector.new(@processed_source.source_buffer, corrections)
+      BASE.new(@processed_source.source_buffer, corrections)
     end
 
-    def diagnostics
-      corrector.diagnostics
+    if defined?(::RuboCop::Cop::Base)
```

Comment:
I'm not sure I understand what's going on. Are you using the existence of `::RuboCop::Cop::Base` to deduce whether `::RuboCop::Cop::Legacy::Corrector` exists?

If that's the case, would it be an option to just check for the existence of `rubocop/cop/legacy/corrector`?

Reply:
> Are you using the existence of `::RuboCop::Cop::Base` to deduce whether `::RuboCop::Cop::Legacy::Corrector` exists?

Yes

> would it be an option to just check for the existence of rubocop/cop/legacy/corrector?

Yes, but I'm not sure I understand how you are proposing to do that exactly, or the benefits.

Reply:
Actually, I'm using it to deduce if `::RuboCop::Cop::Legacy::Corrector` exists *and* which API to use...

Reply:
I'm trying to find out what the actual desired semantics are. As this condition is repeated throughout the diff and you don't seem to be using `::RuboCop::Cop::Base` directly anywhere, I guess you're looking for whether or not the rubocop version used is `0.87+`?

Reply:
Yes, a check on version number would be equivalent

Reply:
I think I'd rather have a check on the version number as that means it'd be more obvious when to remove the added code paths.

But I'm also interested in hearing what @thegedge thinks about this.

Reply:
> it'd be more obvious when to remove the added code paths.

That's a good point 👍

Reply:
Agreed, a version check would definitely make this a little clearer in terms of when we should remove.

Ideally, we would better abstract away the details of Rubocop, but that's outside the scope of this PR. One thing this PR highlights is all the coupling points 🙂

Reply:
Ok, let me switch to version check then.

Reply:
Done